src/main/java/seedu/address/model/profile/Email.java

```diff
@@ -15,20 +15,22 @@ public class Email {
             + "1. The local-part should only contain alphanumeric characters and these special characters, excluding "
             + "the parentheses, (" + SPECIAL_CHARACTERS + "). The local-part may not start or end with any special "
             + "characters.\n"
-            + "2. This is followed by a '@' and then a domain name. The domain name is made up of domain labels "
-            + "separated by periods.\n"
-            + "The domain name must:\n"
-            + "    - end with a domain label at least 2 characters long\n"
-            + "    - have each domain label start and end with alphanumeric characters\n"
-            + "    - have each domain label consist of alphanumeric characters, separated only by hyphens, if any.";
+            + "2. This is followed by a '@' and then a domain name. The domain name must be a valid NUS email domain "
+            + "such as:\n"
+            + "    - u.nus.edu\n"
+            + "    - u.duke.nus.edu\n"
+            + "    - nus.edu.sg\n"
+            + "    - u.yale-nus.edu.sg";
     // alphanumeric and special characters
     private static final String ALPHANUMERIC_NO_UNDERSCORE = "[^\\W_]+"; // alphanumeric characters except underscore
     private static final String LOCAL_PART_REGEX = "^" + ALPHANUMERIC_NO_UNDERSCORE + "([" + SPECIAL_CHARACTERS + "]"
             + ALPHANUMERIC_NO_UNDERSCORE + ")*";
-    private static final String DOMAIN_PART_REGEX = ALPHANUMERIC_NO_UNDERSCORE
-            + "(-" + ALPHANUMERIC_NO_UNDERSCORE + ")*";
-    private static final String DOMAIN_LAST_PART_REGEX = "(" + DOMAIN_PART_REGEX + "){2,}$"; // At least two chars
-    private static final String DOMAIN_REGEX = "(" + DOMAIN_PART_REGEX + "\\.)*" + DOMAIN_LAST_PART_REGEX;
+
+    private static final String STUDENT_EMAIL = "u(\\.duke)?\\.nus\\.edu";
+    private static final String STAFF_EMAIL = "nus\\.edu\\.sg";
+    private static final String YALE_NUS_EMAIL = "u\\.yale-nus\\.edu\\.sg";
```

Comment:
Perhaps it would be better to use a common prefix for the variable naming since these regex are associated. For example, `DOMAIN_STUDENT`, `DOMAIN_STAFF`.

Reply:
Not sure how to account for all NUS emails without listing them exhaustively. If we do choose to restrict to NUS emails, perhaps we can simplify and restrict to only Student and Staff domains which should be sufficient?